meta-balena-common/recipes-support/resin-mounts/resin-mounts/resin-partition-mounter

```diff
@@ -1,121 +1,139 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Tool for mounting a balenaOS partition
+#
+
 set -e
 
-LABEL_TIMEOUT=1900
+TIMEOUT_LOOPS=300 # ~ 100ms/loop * 300loops = 30000ms = 30s
 
+#
+# Help message
+#
 help () {
-	cat << EOF
-Mount a resin partition label
+    cat << EOF
+Tool for mounting a balenaOS partition
 $0 <ARGUMENTS>
 
 ARGUMENTS:
-	-h, --help
-		Print this message.
-	--mount <LABEL>
-		Action: Mount label
-	--umount <LABEL>
-		Action: Unmount label
+    -h, --help
+        Print this message
+    --mount <LABEL>
+        Action: Mount label
+    --umount <LABEL>
+        Action: Unmount label
+    --sysroot
+        Request the action on a sysroot label
 EOF
 }
 
 # Parse arguments
 while [ "$#" -ge 1 ]; do
-	i="$1"
-	case $i in
-		-h|--help)
-			help
-			exit 0
-			;;
-		--mount)
-			if [ -z "$2" ]; then
-				echo "ERROR: $1 argument needs a value."
-			fi
-			part_label="$2"
-			action="mount"
-			shift
-			;;
-		--umount)
-			if [ -z "$2" ]; then
-				echo "ERROR: $1 argument needs a value."
-			fi
-			part_label="$2"
-			action="umount"
-			shift
-			;;
-		--sysroot)
-			sysroot="yes"
-			;;
-		*)
-			echo "ERROR: Unrecognized option $1."
-			help
-			exit 1
-			;;
-	esac
-	shift
+    i="$1"
+    case $i in
+        -h|--help)
+            help
+            exit 0
+            ;;
+        --mount)
+            if [ -z "$2" ]; then
+                echo "ERROR: $1 argument needs a value."
+            fi
+            target="$2"
+            action="mount"
+            shift
+            ;;
+        --umount)
+            if [ -z "$2" ]; then
+                echo "ERROR: $1 argument needs a value."
+            fi
+            target="$2"
+            action="umount"
+            shift
+            ;;
+        --sysroot)
+            sysroot="yes"
+            ;;
+        *)
+            echo "ERROR: Unrecognized option $1."
+            help
+            exit 1
+            ;;
+    esac
+    shift
 done
 
 if [ -z "$action" ]; then
-	echo "ERROR: Required arguments not provided."
-	help
-	exit 1
+    echo "ERROR: Required arguments not provided."
+    help
+    exit 1
 fi
 
+# shellcheck disable=SC1091
+. /usr/libexec/os-helpers-fs
+
+# Figure out what (target_path) and where (target_mountpoint)
 if [ -z "$sysroot" ]; then
-	part_mountpoint="/mnt/${part_label#resin-}"
-	label_path="/dev/disk/by-label/$part_label"
+    target_mountpoint="/mnt/${target#resin-}"
+    target_path="$(get_state_path_from_label "$target")"
 else
-	part_mountpoint="/mnt/sysroot/$part_label"
-	label_path="/dev/disk/by-state/$part_label"
+    target_mountpoint="/mnt/sysroot/$target"
+    target_path="/dev/disk/by-state/$target"
```

Comment:
I'm wondering if the fallback logic can all be in one place instead of separately.
i.e. get path. and if label path exists fine, else return the fallback.

Reply:
Not sure I follow, isn't that what the get_state_path_from_label() does:
```
 63 get_state_path_from_label() {                                                   
 64         [ -z "$1" ] && return 1                                                 
 65         if [ -L "/dev/disk/by-state/${1#resin-}" ] ; then                       
 66                 echo "/dev/disk/by-state/${1#resin-}"                           
 67         else                                                                    
 68                 echo "/dev/disk/by-label/$1"                                    
 69         fi                                                                      
 70 }  
```
if the state symlink exists it uses it, otherwise it uses the label symlink.
Are you referring to something else?

Reply:
As per the latest arch discussion, I will remove the fallback to labels as this may hide issues and there is no real use case for it.